src/Diagnostics/CodeAnalysis/Core/MetaAnalyzers/RegisterActionAnalyzer.cs

```diff
@@ -245,10 +371,172 @@ protected override void AnalyzeNode(SymbolAnalysisContext symbolContext, TInvoca
                         }
 
                         var diagnostic = Diagnostic.Create(InvalidSyntaxKindTypeArgumentRule, location, typeArgument.Name, TLanguageKindEnumName, method.Name);
-                        symbolContext.ReportDiagnostic(diagnostic);
+                        context.ReportDiagnostic(diagnostic);
                     }
                 }
             }
+
+            private void AnalyzeArgumentSyntax(SyntaxNodeAnalysisContext context)
+            {
+                var argumentExpression = GetArgumentExpression((TArgumentSyntax)context.Node);
+                if (argumentExpression != null)
+                {
+                    var parameter = context.SemanticModel.GetSymbolInfo(argumentExpression, context.CancellationToken).Symbol as IParameterSymbol;
+                    if (parameter != null)
+                    {
+                        AnalyzeParameterReference(parameter);
+                    }
+                }
+            }
+
+            private void AnalyzerParameterSyntax(SyntaxNodeAnalysisContext context)
+            {
+                var parameter = context.SemanticModel.GetDeclaredSymbol(context.Node, context.CancellationToken) as IParameterSymbol;
+                if (parameter != null)
+                {
+                    AnalyzeParameterDeclaration(parameter);
+                }
+            }
+
+            private void AnalyzeParameterDeclaration(IParameterSymbol parameter)
+            {
+                if (IsContextType(parameter.Type, _compilationStartAnalysisContext, _codeBlockStartAnalysisContext))
+                {
+                    _declaredStartAnalysisContextParams = _declaredStartAnalysisContextParams ?? new HashSet<IParameterSymbol>();
+                    _declaredStartAnalysisContextParams.Add(parameter);
+                }
+            }
+
+            private void AnalyzeParameterReference(IParameterSymbol parameter)
+            {
+                if (IsContextType(parameter.Type, _compilationStartAnalysisContext, _codeBlockStartAnalysisContext))
+                {
+                    _startAnalysisContextParamsToSkip = _startAnalysisContextParamsToSkip ?? new HashSet<IParameterSymbol>();
+                    _startAnalysisContextParamsToSkip.Add(parameter);
```

Comment:
I know there is an explanatory comment on the declaration of the collection, but consider adding a comment explaining why this parameter is to be skipped.